pkg/utils/secrets/certificate.go

```diff
@@ -309,3 +314,8 @@ func SelfGenerateTLSServerCertificate(name string, dnsNames []string, ips []net.
 
 	return certificate, caCertificate, tempDir, nil
 }
+
+// AdjustToClockSkew adjusts the given time by the maximum allowed clock skew.
```

Comment:
```suggestion
// AdjustToClockSkew adjusts the given time by the maximum allowed clock skew. It is done to prevent confusing non-transient errors
```
If clock skew can cause non-transient errors I think adding this gives better context and justification